doc/specs/stdlib_stringlist.md

```diff
@@ -0,0 +1,467 @@
+If the value is lower than that of `first`, the range is considered to be empty and nothing is deleted.
+
+
+### `destroy` - destroy all strings in the list
+
+#### Status
+
+Experimental
+
+#### Description
+
+Destroy the entire contents of the list. As the variable holding the list is simply a derived type, the variable
+itself is not destroyed.
+
+#### Syntax
+
+`call list % [[stringlist_type(type):destroy(bound)]]`
+
+#### Class
+
+Subroutine
+
+#### Arguments
+
+`list`: the stringlist variable from which to delete all strings
+
+
+### `get` - get a single string from the list
+
+#### Status
+
+Experimental
+
+#### Description
+
+Get the string at the given position.
+
+#### Syntax
+
+`string = list % [[stringlist_type(type):get(bound) ( idx )]]`
+
+#### Class
+
+Function
+
+#### Arguments
+
+`list`: the stringlist variable to retrieve a string from
+
+`idx`: the index of the string to be retrieved (see [`the section on positions`](./stdlib_stringlist.html#position-in-a-list-of-strings)
+
+#### Result value
+
+A copy of the string stored at the indicated position.
+
+
+### `index` - find the index of a particular string in the list
+
+#### Status
+
+Experimental
+
+#### Description
+
+Get the position of the first stored string that matches the given string, if `back` is not present or false. If `back` is
+false, return the position of the last stored string that matches. Note that trailing blanks are ignored.
+
+#### Syntax
+
+`idx = list % [[stringlist_type(type):index(bound) ( string, back )]]`
+
+#### Class
+
+Function
+
+#### Arguments
+
+`list`: the stringlist variable to retrieve a string from
+
+`string`: the string to be found in the list
+
+`back` (optional): logical argument indicating the first occurrence should be returned (`false`) or the last (`true`)
+
+#### Result value
+
+The result is either the index of the string in the list or -1 if the string was not found
+
+#### Example
+
+Because trailing blanks are ignored, the following calls will give the same result:
+
+```fortran
+    write(*,*) list%index( 'A' )
+    write(*,*) list%index( 'A    ' )
+```
+
+
+### `index_sub` - find the index of a string containing the given substring in the list
+
+#### Status
+
+Experimental
+
+#### Description
+
+Get the position of the first stored string that contains the given substring, if `back` is not present or false. If `back` is
+false, return the position of the last stored string that contains it.
+
+#### Syntax
+
+`idx = list % [[stringlist_type(type):index_sub(bound) ( substring, back )]]`
+
+#### Class
+
+Function
+
+#### Arguments
+
+`list`: the stringlist variable to retrieve a string from
+
+`substring`: the substring in question
+
+`back` (optional): logical argument indicating the first occurrence should be returned (`false`) or the last (`true`)
+
+#### Result value
+
+The result is either the index of the string in the list or -1 if the string was not found
```

Comment:
Same question on the return code as with `index`.

Reply:
Indeed